internal/codespaces/api/api.go

```diff
@@ -686,3 +689,19 @@ func (a *API) do(ctx context.Context, req *http.Request, spanName string) (*http
 func (a *API) setHeaders(req *http.Request) {
 	req.Header.Set("Accept", "application/vnd.github.v3+json")
 }
+
+// withRetry takes a generic function that sends an http request and retries
+// only when the returned response has a >=500 status code.
+func (a *API) withRetry(f func() (*http.Response, error)) (resp *http.Response, err error) {
+	for i := time.Duration(0); i < 5; i++ {
+		resp, err = f()
+		if err != nil {
+			return nil, err
+		}
+		if resp.StatusCode >= 500 {
+			time.Sleep(a.retryBackoff * (i + 1))
+			continue
```

Comment:
I think you forgot to break out of the loop in the "else" case.

Stylistically, it's best to put the break under the condition and let the continue be implicit:

```
if resp.StatusCode < 500 {
	break
}
time.Sleep(...)
```

Reply:
Thanks for the catch. Added a test to ensure that retries == 0 if the call immediately succeeds.